src/interpreter/index.ts

```diff
@@ -122,7 +122,7 @@ export class Interpreter implements Expr.Visitor<BrsType>, Stmt.Visitor<BrsType>
                 );
                 let statements = await componentScopeResolver.resolve(component);
                 interpreter.inSubEnv(subInterpreter => {
-                    subInterpreter.environment.setM(interpreter.environment.getM());
+                    subInterpreter.environment.setM(new RoAssociativeArray([]));
```

Comment:
One thing I have tried that _seems to work_ is also setting the environments rootM here to the new associative array and then revevrting the below change to call `getRootM` again